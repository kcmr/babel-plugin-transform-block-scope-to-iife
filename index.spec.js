const path = require('path');
const babel = require('@babel/core');
const plugin = require('.');

const transformFixture = (fixture, options = {}) => {
  const fixturePath = path.join(__dirname, '__fixtures__', fixture);
  const { code } = babel.transformFileSync(fixturePath, {
    plugins: [[plugin, options]],
  });

  return code;
};

describe('babel-plugin-transform-block-scope-to-iife', () => {
  it('transforms top level block scope by an IIFE', () => {
    expect(transformFixture('top-level-block-scope.js')).toMatchSnapshot();
  });
});
