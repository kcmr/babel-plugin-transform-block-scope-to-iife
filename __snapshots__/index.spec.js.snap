// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-transform-block-scope-to-iife transforms top level block scope by an IIFE 1`] = `
"(function () {
  const {
    Element,
    html
  } = window.Polymer;

  class Component extends Element {
    static get is() {
      return 'tag-name';
    }

  }
})();

class Component extends Element {
  static get is() {
    return 'tag-name';
  }

}

if (true) {// code
}"
`;
